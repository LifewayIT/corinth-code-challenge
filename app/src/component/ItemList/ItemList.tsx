import {Avatar, List, ListItem, ListItemAvatar, ListItemText, ListSubheader} from '@mui/material';
import {isEmpty} from 'lodash';
import React, {ReactElement} from 'react';
import {Link} from 'react-router-dom';
import {toResourceUrl} from '../../util/data';

type GraphQLType = {
    __typename?: string;
    name: string;
    picture?: string;
    id: string;
};

interface ItemListProps<T extends GraphQLType> {
    items: T[];
    showType?: boolean;
    title?: string;
}

export const ItemList =
    <T extends GraphQLType>({items, showType = false, title}: ItemListProps<T>): ReactElement =>
        !isEmpty(items) ? (
            <List dense
                  sx={{
                      'width': '100%',
                      'maxWidth': 360,
                      'bgcolor': 'background.paper',
                      'position': 'relative',
                      'overflow': 'auto',
                      'maxHeight': 200,
                      '& ul': {padding: 0},
                  }}
                  subheader={title && (
                      <ListSubheader component="div" id="nested-list-subheader" color="primary">
                          {title}
                      </ListSubheader>
                  )}>
                {items.map((value) => (
                    <Link to={{
                        pathname: toResourceUrl(value)
                    }} key={`search-result-${value.id}`}>
                        <ListItem button dense>
                            <ListItemAvatar>
                                <Avatar src={value.picture}/>
                            </ListItemAvatar>
                            <ListItemText primary={value.name}
                                          secondary={showType && value.__typename}/>
                        </ListItem>
                    </Link>
                ))}
            </List>
        ) : (<></>);
