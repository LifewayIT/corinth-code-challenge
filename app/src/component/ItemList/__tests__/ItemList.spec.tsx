import React from 'react';
import {create} from 'react-test-renderer';
import {ItemList} from '../ItemList';

describe('ItemList', () => {

    it('should render without error', () => {

        const tree = create(
            <ItemList items={[]}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();

    });

});
