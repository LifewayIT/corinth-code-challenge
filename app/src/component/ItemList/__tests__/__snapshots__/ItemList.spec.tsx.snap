// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList should render without error 1`] = `null`;
